### PR TITLE
feat: re enable partial loading for ABs

### PR DESCRIPTION
**Pull request by davidejensen**
# Pull Request Description

## What does this PR change?
This PR re-enables the partial loading flow for asset bundles with the improvements on the memory allocations side.
Nick introduced a slab allocator that allows us to process initial chunks of partial files without the need of already allocating the full file required space

### Prerequisites
- [ ] List any required setup steps
- [ ] Include environment/configuration requirements

### Test Steps
1. First step
2. Second step
3. Expected result after step 2
4. ...

### Additional Testing Notes
- Note any edge cases to verify
- Mention specific areas that need careful testing
- List known limitations or potential issues

## Quality Checklist
- [ ] Changes have been tested locally
- [ ] Documentation has been updated (if required)
- [ ] Performance impact has been considered
- [ ] For SDK features: Test scene is included

## Code Review Reference
Please review our [Code Review Standards](https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md) before submitting.

### CODE REVIEW

**Comment by AnsisMalins** [comment]
> without the need of already allocating the full file required space

But that is not a good thing. If you don't have enough disk space, you should find out immediately and not after you have downloaded half a bundle already.

File: Explorer/Assets/Scripts/Utility/Multithreading/Mutex.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
+using Cysharp.Threading.Tasks;
+using System;
+using System.Threading;
+
+namespace Utility.Multithreading
+{
+    public class MutexSlim<T> : IDisposable where T: IDisposable
```
Comment (AnsisMalins) [inline Explorer/Assets/Scripts/Utility/Multithreading/Mutex.cs:7]: What does this do that `lock` does not?

**Review by AnsisMalins — CHANGES_REQUESTED** [review]
I don't like it. But I also don't fully understand it.

**Comment by mikhail-dcl** [comment]
Superseded by #3965
